IBPSA/Fluid/Sources/Boundary_pT.mo

```diff
@@ -1,95 +1,9 @@
-  parameter Medium.ExtraProperty C[Medium.nC](
-    final quantity=Medium.extraPropertiesNames) = fill(0, Medium.nC)
-    "Fixed values of trace substances"
-    annotation (Dialog(enable = (not use_C_in) and Medium.nC > 0));
-  Modelica.Blocks.Interfaces.RealInput p_in(final unit="Pa") if use_p_in
-    "Prescribed boundary pressure"
-    annotation (Placement(transformation(extent={{-140,60},{-100,100}})));
-  Modelica.Blocks.Interfaces.RealInput T_in(final unit="K",
-                                            displayUnit="degC") if use_T_in
-    "Prescribed boundary temperature"
-    annotation (Placement(transformation(extent={{-140,20},{-100,60}})));
-  Modelica.Blocks.Interfaces.RealInput X_in[Medium.nX](
-    each final unit = "kg/kg",
-    final quantity=Medium.substanceNames) if use_X_in
-    "Prescribed boundary composition"
-    annotation (Placement(transformation(extent={{-140,-60},{-100,-20}})));
-  Modelica.Blocks.Interfaces.RealInput C_in[Medium.nC](
-    final quantity=Medium.extraPropertiesNames) if use_C_in
-    "Prescribed boundary trace substances"
-    annotation (Placement(transformation(extent={{-140,-100},{-100,-60}})));
-protected
-  Modelica.Blocks.Interfaces.RealInput p_in_internal(final unit="Pa")
-    "Needed to connect to conditional connector";
-  Modelica.Blocks.Interfaces.RealInput T_in_internal(final unit="K",
-                                                     displayUnit="degC")
-    "Needed to connect to conditional connector";
-  Modelica.Blocks.Interfaces.RealInput X_in_internal[Medium.nX](
-    each final unit = "kg/kg",
-    final quantity=Medium.substanceNames)
-    "Needed to connect to conditional connector";
-  Modelica.Blocks.Interfaces.RealInput C_in_internal[Medium.nC](
-    final quantity=Medium.extraPropertiesNames)
-    "Needed to connect to conditional connector";
-
-initial equation
-  if not use_X_in then
-    Modelica.Fluid.Utilities.checkBoundary(Medium.mediumName, Medium.substanceNames,
-      Medium.singleState, true, X_in_internal, "Boundary_pT");
-  end if;
-
-equation
-  if use_X_in then
-    Modelica.Fluid.Utilities.checkBoundary(Medium.mediumName, Medium.substanceNames,
-      Medium.singleState, true, X_in_internal, "Boundary_pT");
-  end if;
-  connect(p_in, p_in_internal);
-  connect(T_in, T_in_internal);
-  connect(X_in, X_in_internal);
-  connect(C_in, C_in_internal);
-  if not use_p_in then
-    p_in_internal = p;
-  end if;
-  if not use_T_in then
-    T_in_internal = T;
-  end if;
-  if not use_X_in then
-    X_in_internal = X;
-  end if;
-  if not use_C_in then
-    C_in_internal = C;
-  end if;
-  medium.p = p_in_internal;
-  medium.T = T_in_internal;
-  medium.Xi = X_in_internal[1:Medium.nXi];
-  ports.C_outflow = fill(C_in_internal, nPorts);
+  extends IBPSA.Fluid.Sources.BaseClasses.PartialSource_p;
+  extends IBPSA.Fluid.Sources.BaseClasses.PartialSource_T;
+  extends IBPSA.Fluid.Sources.BaseClasses.PartialSource_Xi_C;
   annotation (defaultComponentName="bou",
```

Comment:
You will also need to move the graphical annotation such as
```
        Text(
          visible=use_C_in,
          extent={{-164,-90},{-62,-130}},
          lineColor={0,0,0},
          fillColor={255,255,255},
          fillPattern=FillPattern.Solid,
          textString="C"),
...
        Line(
          visible=use_C_in,
          points={{-100,-80},{-60,-80}},
          color={0,0,255}),
```
to the respective base class.

Reply:
Okay, since `Boundary_p*` and `MassFlowSource_*` now share the same code, I removed the lines that connect the inputs to the model icon, since the model icons have a different shape.